internal/telemetry/segment.go

```diff
@@ -172,3 +174,13 @@ type body struct {
 	Timestamp  string            `json:"timestamp"`
 	WriteKey   string            `json:"writeKey"`
 }
+
+func findErrorCode(properties map[string]string, err error) {
+	if v, ok := err.(interface{ErrorCode() string}); ok {
```

Comment:
Should we introduce this as a formal interface?

Reply:
🤷 this is a pretty standard Go pattern. You want a public interface?